src/auto/app_info.rs

```diff
@@ -80,15 +84,26 @@ impl AppInfo {
         }
     }
 
-    //#[cfg(any(feature = "v2_50", feature = "dox"))]
-    //pub fn launch_default_for_uri_async<'a, 'b, P: Into<Option<&'a Cancellable>>, Q: Into<Option<&'b /*Unimplemented*/AsyncReadyCallback>>, R: Into<Option</*Unimplemented*/Fundamental: Pointer>>>(uri: &str, launch_context: &AppLaunchContext, cancellable: P, callback: Q, user_data: R) {
-    //    unsafe { TODO: call ffi::g_app_info_launch_default_for_uri_async() }
-    //}
-
-    //#[cfg(any(feature = "v2_50", feature = "dox"))]
-    //pub fn launch_default_for_uri_finish<P: IsA</*Ignored*/AsyncResult>>(result: &P) -> Result<(), Error> {
-    //    unsafe { TODO: call ffi::g_app_info_launch_default_for_uri_finish() }
-    //}
+    #[cfg(any(feature = "v2_50", feature = "dox"))]
+    pub fn launch_default_for_uri_async<P: Fn(Result<(), Error>) + Send + Sync>(uri: &str, launch_context: &AppLaunchContext, callback: P) -> Cancellable {
+        let cancellable = Cancellable::new();
```

Comment:
There can be no Cancellable, or the same Cancellable could be used for multiple operations. Maybe let the user allocate it if needed (it's not cheap) and let the user pass an `Option<&Cancellable>`?

Reply:
By "There can be no Cancellable", you mean the user could want no `Cancellable` and you want to have the same API as in C, meaning that the user decide whether there is a `Cancellable` or not?

Reply:
Yes. The user could create a `Cancellable` or pass `None`. And the user could use the same `Cancellable` for multiple operations, even simultaneously or one after another.